hazelcast/src/main/java/com/hazelcast/wan/impl/WanReplicationService.java

```diff
@@ -208,4 +210,10 @@ DistributedServiceWanEventCounters getSentEventCounters(String wanReplicationNam
      * @param dataStructureName the distributed object name
      */
     void removeWanEventCounters(String serviceName, String dataStructureName);
+
+    /**
+     * Returns an immutable collection of all WAN protocol versions supported by
+     * this instance.
+     */
+    Collection<Version> getSupportedWanProtocolVersions();
```

Comment:
Probably use a `List` here, so that versions are sorted preferred-first?

Reply:
Right, I realised this also when re-reading the research. Will change.